include/pybind11/pybind11.h

```diff
@@ -41,6 +41,11 @@
 #  endif
 #endif
 
+#include <memory>
+#include <vector>
+#include <string>
+#include <utility>
+
```

Comment:
Before or after the other pybind11 headers? I tend to include stdlib as last includes, but I'm not sure if there's a policy or habit, here. Apart from that, fine with me! :-)

Reply:
I can change the order, if you like? :)

Reply:
Here: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes
I read: 

_Include headers in the following order: Related header, C system headers, C++ standard library headers, other libraries' headers, your project's headers._

But that is Googles rules, not necessarily yours.

Reply:
I think LLVM and therefore the `clang-*` tools have the same ordering recommendation.

Reply:
PS: There is one hard requirement for all Python modules, though: `Python.h` must precede all stdlib headers.

Reply:
Also fine to keep it this way, then. Just wanted to open this for discussion, and check whether this is what we want :-)

Reply:
Actually, I think this has to be moved. This puts stdlib includes above `<Python.h>`, which causes massive numbers of warnings on Python 2.7 on older Linux systems (at least). The rule is `Python.h` must be above any stdlib includes for any extension module, and that rule is kept by by pybind11.

Reply:
Ok! I will do another commit soon, within the hour, moving them under the project headers!